impexp-client/src/main/java/org/citydb/gui/modules/common/filter/XMLQueryView.java

```diff
@@ -463,14 +467,22 @@ public Icon getIcon() {
 
     @Override
     public void loadSettings() {
-        QueryConfig query = config.getExportConfig().getQuery();
-        xmlText.setText(marshalQuery(query, config.getNamespaceFilter()));
+        QueryConfig query = queryConfigSupplier.get();
+        xmlText.setText(marshalQuery(query, configNamespaceFilterSupplier.get()));
```

Comment:
I think we should only require suppliers for settings that are specific to the "user" of the filter view (for instance, "user" = export or delete operation).

The `ConfigNamespaceFilter` is not user-specific. It is created when loading the `Config` and only changes when loading a new config. So, I recommend to use `ConfigNamespaceFilter namespaceFilter = ObjectRegistry.getInstance().getConfig().getNamespaceFilter();` instead.

Reply:
Agree. Done in commit cec12f3